rllib/policy/torch_policy.py

```diff
@@ -641,12 +641,11 @@ def learn_on_loaded_batch(self, offset: int = 0, buffer_index: int = 0):
         self.apply_gradients(_directStepOptimizerSingleton)
 
         batch_fetches = {}
+        batch_fetches[LEARNER_STATS_KEY] = {}
         for i, batch in enumerate(device_batches):
-            batch_fetches[f"tower_{i}"] = {
-                LEARNER_STATS_KEY: self.extra_grad_info(batch)
-            }
+            batch_fetches[LEARNER_STATS_KEY][f"tower_{i}"] = self.extra_grad_info(batch)
 
-        batch_fetches.update(self.extra_compute_grad_fetches())
+        batch_fetches[LEARNER_STATS_KEY].update(self.extra_compute_grad_fetches())
```

Comment:
@Bam4d,
Nice. What do you think of using merge_dicts which will merge them recursively.

https://github.com/ray-project/ray/blob/d9552e67950c0211d727a11873a5c95dd510108f/python/ray/util/ml_utils/dict.py#L9-L20

```python
from ray.rllib.utils import merge_dicts

batch_fetches = merge_dicts(batch_fetches, self.extra_compute_grad_fetches(), True)
```

Reply:
I think this is probably a better way of doing it thanks!